docs/src/doc/user-guide/advanced/graal-vm-native-image.md

```diff
@@ -17,4 +21,24 @@ godot {
 ```
 
 In order to use the generated native-image, you can pass `--java-vm-type=graal` argument to engine, or simply change
-`godot_kotlin_configuration.json` to set vm_type to `graal`.
\ No newline at end of file
+`godot_kotlin_configuration.json` to set vm_type to `graal`.
+
+## Reflexion and JNI with native image
+
+GraalVM native image performs AOT compilation. In order to be able to use reflexion and jni, you need to provide an
+additional configuration file.  
+This applies also for third party library you use that would do reflexion.
+
+In order to append those configurations add the json in `graal` folder of your project (it should be generated on first
```

Comment:
I would add a link to the graalvm docs on how to generate these configs. Atm this part is only clear to those who already built native images in the past.